src/Blocks/custom/carousel/assets/carousel-slider.js

```diff
@@ -0,0 +1,68 @@
+import Swiper, { Navigation, Pagination } from 'swiper';
+
+Swiper.use([Navigation, Pagination]);
+
+export class CarouselSlider {
+	constructor(options) {
+		this.element = options.element;
+		this.blockClass = options.blockClass;
+		this.nextElement = options.nextElement;
+		this.prevElement = options.prevElement;
+		this.paginationElement = options.paginationElement;
+		this.eventName = options.eventName;
+		this.swipeHandler = options.eventName;
+	}
+
+	init() {
+		const item = this.element;
+
+		new Swiper(item, {
+			loop: item.getAttribute('data-swiper-loop'),
+			slideClass: `${this.blockClass}__item`,
+			slidesPerView: 2,
+			spaceBetween: 30,
+			keyboard: {
+				enabled: true,
+			},
+			swipeHandler: this.swipeHandler,
+			centeredSlides: true,
+			grabCursor: false,
+			breakpointsInverse: true,
+			threshold: 20,
+			navigation: {
+				nextEl: this.nextElement,
+				prevEl: this.prevElement,
+			},
+			pagination: {
+				el: this.paginationElement,
+				type: 'fraction',
+				renderFraction: (currentClass, totalClass) => {
+					return `<span class="${currentClass}"></span><span>/</span><span class="${totalClass}"></span>`;
+				},
+			},
+			on: {
+				init: () => {
+					window.dispatchEvent(this.eventName);
+				},
+			},
+			breakpoints: {
+				// when width >= 1px
+				1: {
```

Comment:
you can put this outside of the breakepoints

Reply:
you resolved the comment but haven't fixed the issue?

Reply:
Sorry, did you mean that I should move the comment or?

If the comment was the issue, that's fixed in the semi-recent push.

Reply:
The issue here is breakpoint on 1 px. In Swiper if you put `breakpointsInverse: true` this means that you are doing mobile first approach. Then you dont need this 1 brake point.

```js
new Swiper(
	item, {
	...
	slidesPerView: 1,
	....
	breakpoints: {
		849: {
			slidesPerView: 2,
		},
		1470: {
			slidesPerView: 3,
		},
		2001: {
			slidesPerView: 5,
		},
	}
);
```